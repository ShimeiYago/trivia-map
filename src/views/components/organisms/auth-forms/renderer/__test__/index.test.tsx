import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, Renderer>;

const basicProps: Props = {
  autoLoggingInState: 'waiting',
  initialMode: 'login',
  loginSuccess: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('signup mode', () => {
    wrapper.setState({
      mode: 'signup',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('reset-pasword mode', () => {
    wrapper.setState({
      mode: 'reset-password',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('resend-email mode', () => {
    wrapper.setState({
      mode: 'resend-email',
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('switchMode', () => {
  it('should provide a handler to change mode', () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    instance['switchMode']('signup')();

    expect(instance.state.mode).toBe('signup');
  });
});

describe('handleChangeEmail', () => {
  it('should change email state', () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    const inputEvent = {
      target: {
        value: 'xxx@example.com',
      },
    } as React.ChangeEvent<HTMLInputElement>;

    instance['handleChangeEmail'](inputEvent);

    expect(instance.state.email).toBe('xxx@example.com');
  });
});
