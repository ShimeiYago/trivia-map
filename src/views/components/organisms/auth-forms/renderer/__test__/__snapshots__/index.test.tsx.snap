// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Box)
  maxWidth="340px"
  sx={
    Object {
      "mx": "auto",
    }
  }
>
  <LoginForm
    autoLoggingInState="waiting"
    email=""
    loginSuccess={[MockFunction]}
    onChangeEmail={[Function]}
    setAccessTokenExpiration={[MockFunction]}
    setRefreshTokenExpiration={[MockFunction]}
    switchMode={[Function]}
  />
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests resend-email mode 1`] = `
<ForwardRef(Box)
  maxWidth="340px"
  sx={
    Object {
      "mx": "auto",
    }
  }
>
  <ResendEmailForm
    email=""
    onChangeEmail={[Function]}
    switchMode={[Function]}
  />
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests reset-pasword mode 1`] = `
<ForwardRef(Box)
  maxWidth="340px"
  sx={
    Object {
      "mx": "auto",
    }
  }
>
  <PasswordResetRequestForm
    email=""
    onChangeEmail={[Function]}
    switchMode={[Function]}
  />
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests signup mode 1`] = `
<ForwardRef(Box)
  maxWidth="340px"
  sx={
    Object {
      "mx": "auto",
    }
  }
>
  <SignupForm
    email=""
    onChangeEmail={[Function]}
    switchMode={[Function]}
  />
</ForwardRef(Box)>
`;
